frontend/src/lib/StaticGraphParser.ts

```diff
@@ -15,196 +15,184 @@
+    return info;
+  }
+
+  info.nodeType = 'container';
+  info.args = template.container.args || [],
+  info.command = template.container.command || [],
+  info.image = template.container.image || '';
+
+  if (template.inputs) {
+    info.inputs =
+      (template.inputs.parameters || []).map(p => [p.name, p.value || '']);
+  }
+  if (template.outputs) {
+    info.outputs = (template.outputs.parameters || []).map(p => {
+      let value = '';
+      if (p.value) {
+        value = p.value;
+      } else if (p.valueFrom) {
+        value = p.valueFrom.jqFilter || p.valueFrom.jsonPath || p.valueFrom.parameter || p.valueFrom.path || '';
+      }
+      return [p.name, value];
+    });
+  }
+  return info;
+}
+
+/**
+ * Recursively construct the static graph of the Pipeline.
+ *
+ * @param graph The dagre graph object
+ * @param rootTemplateId The current template being explored at this level of recursion
+ * @param templates An unchanging map of template name to template and type
+ * @param parent The task that was being examined when the function recursed
+ */
+function buildDag(
+    graph: dagre.graphlib.Graph,
+    rootTemplateId: string,
+    templates: Map<string, { nodeType: nodeType, template: Template }>,
+    parent?: string): void {
+
+  const root = templates.get(rootTemplateId);
+
+  if (root && root.nodeType === 'dag') {
+    const template = root.template;
+
+    (template.dag.tasks || []).forEach((task) => {
+      // If the user specifies an exit handler, then the compiler will wrap the entire Pipeline
+      // within an additional DAG template prefixed with "exit-handler".
+      // If this is the case, we simply treat it as the root of the graph and work from there
+      if (task.name.startsWith('exit-handler')) {
+        buildDag(graph, task.template, templates);
+      } else {
+
+        // Parent here will be the task that pointed to this DAG template
+        if (parent) {
+          graph.setEdge(parent, task.name);
+        }
+
+        // This object contains information about the node that we display to the user when they
+        // click on a node in the graph
+        const info = new SelectedNodeInfo();
+        if (task.when) {
+          info.condition = task.when;
+        }
+
+        // "Child" here is the template that this task points to. This template should either be a
+        // DAG, in which case we recurse, or a container which can be thought of as a leaf node
+        const child = templates.get(task.template);
+        if (child) {
+          if (child.nodeType === 'dag') {
+            buildDag(graph, task.template, templates, task.name);
+          } else if (child.nodeType === 'container' ) {
+            _populateInfoFromTemplate(info, child.template);
+          } else {
+            logger.error(`Unknown nodetype: ${child.nodeType} on template: ${child.template}`);
+          }
+        }
+
+        graph.setNode(task.name, {
+          bgColor: task.when ? 'cornsilk' : undefined,
```

Comment:
Nit: This should probably be performed using a CSS class.

Reply:
Agreed. This styling is still technically under discussion, so I'll leave it for now, but once it's decided (if we're keeping it), I'll move it to somewhere more appropriate